gst-ffmpeg: ffserver: set oformat

Fix Ticket1986

Signed-off-by: Michael Niedermayer <michaelni@gmx.at>
(cherry picked from commit cbe43e62c9ac7d4aefdc13476f6f691bd626525f)

Upstream-Status: Backport 

---
 ffserver.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/ffserver.c b/ffserver.c
index 4044d0f..8740140 100644
--- a/gst-libs/ext/libav/ffserver.c
+++ b/gst-libs/ext/libav/ffserver.c
@@ -2937,12 +2937,14 @@ static int prepare_sdp_description(FFStream *stream, uint8_t **pbuffer,
 {
     AVFormatContext *avc;
     AVStream *avs = NULL;
+    AVOutputFormat *rtp_format = av_guess_format("rtp", NULL, NULL);
     int i;
 
     avc =  avformat_alloc_context();
-    if (avc == NULL) {
+    if (avc == NULL || !rtp_format) {
         return -1;
     }
+    avc->oformat = rtp_format;
     av_dict_set(&avc->metadata, "title",
                stream->title[0] ? stream->title : "No Title", 0);
     avc->nb_streams = stream->nb_streams;
-- 
1.7.5.4

